cranelift-codegen/meta/src/shared/legalize.rs

```diff
@@ -272,48 +289,92 @@ pub(crate) fn define(insts: &InstructionGroup, imm: &Immediates) -> TransformGro
         ],
     );
 
-    let e_eq = Literal::enumerator_for(&imm.intcc, "eq");
+    let intcc_eq = Literal::enumerator_for(&imm.intcc, "eq");
     narrow.legalize(
-        def!(b = icmp_imm.I64(e_eq, x, y)),
+        def!(b = icmp_imm.I64(intcc_eq, x, y)),
         vec![
             def!((xl, xh) = isplit(x)),
-            def!(b1 = icmp_imm(e_eq, xl, y)),
-            def!(b2 = icmp_imm(e_eq, xh, y)),
+            def!(b1 = icmp_imm(intcc_eq, xl, y)),
+            def!(b2 = icmp_imm(intcc_eq, xh, y)),
             def!(b = band(b1, b2)),
         ],
     );
     narrow.legalize(
-        def!(b = icmp.I64(e_eq, x, y)),
+        def!(b = icmp.I64(intcc_eq, x, y)),
         vec![
             def!((xl, xh) = isplit(x)),
             def!((yl, yh) = isplit(y)),
-            def!(b1 = icmp(e_eq, xl, yl)),
-            def!(b2 = icmp(e_eq, xh, yh)),
+            def!(b1 = icmp(intcc_eq, xl, yl)),
+            def!(b2 = icmp(intcc_eq, xh, yh)),
             def!(b = band(b1, b2)),
         ],
     );
 
-    let e_ne = Literal::enumerator_for(&imm.intcc, "ne");
+    let intcc_ne = Literal::enumerator_for(&imm.intcc, "ne");
     narrow.legalize(
-        def!(b = icmp_imm.I64(e_ne, x, y)),
+        def!(b = icmp_imm.I64(intcc_ne, x, y)),
         vec![
             def!((xl, xh) = isplit(x)),
-            def!(b1 = icmp_imm(e_ne, xl, y)),
-            def!(b2 = icmp_imm(e_ne, xh, y)),
+            def!(b1 = icmp_imm(intcc_ne, xl, y)),
+            def!(b2 = icmp_imm(intcc_ne, xh, y)),
             def!(b = band(b1, b2)),
         ],
     );
     narrow.legalize(
-        def!(b = icmp.I64(e_ne, x, y)),
+        def!(b = icmp.I64(intcc_ne, x, y)),
         vec![
             def!((xl, xh) = isplit(x)),
             def!((yl, yh) = isplit(y)),
-            def!(b1 = icmp(e_ne, xl, yl)),
-            def!(b2 = icmp(e_ne, xh, yh)),
+            def!(b1 = icmp(intcc_ne, xl, yl)),
+            def!(b2 = icmp(intcc_ne, xh, yh)),
             def!(b = bor(b1, b2)),
         ],
     );
 
+    for cc in &["sgt", "sge", "slt", "sle", "ugt", "uge", "ult", "ule"] {
+        let intcc_cc = Literal::enumerator_for(&imm.intcc, cc);
+
+        let cc1 =
+            Literal::enumerator_for(&imm.intcc, Box::leak(without_equal(cc).into_boxed_str()));
+        let cc2 = Literal::enumerator_for(
+            &imm.intcc,
+            Box::leak(inverse_without_equal(cc).into_boxed_str()),
```

Comment:
Maybe add a method to IntCC returning `&'static str` instead, and use it in the Disply impl of IntCC too.